src/ESPEasyRules.ino

```diff
@@ -156,42 +156,83 @@ String rulesProcessingFile(const String& fileName, String& event) {
   fs::File f = tryOpenFile(fileName, "r+");
   SPIFFS_CHECK(f, fileName.c_str());
 
-  String line      = "";
-  bool   match     = false;
-  bool   codeBlock = false;
-  bool   isCommand = false;
-  bool   condition[RULES_IF_MAX_NESTING_LEVEL];
-  bool   ifBranche[RULES_IF_MAX_NESTING_LEVEL];
-  byte   ifBlock     = 0;
-  byte   fakeIfBlock = 0;
-
-  byte *buf = new byte[RULES_BUFFER_SIZE]();
+  String line;
+  line.reserve(RULES_IF_MAX_NESTING_LEVEL);
+  bool match     = false;
+  bool codeBlock = false;
+  bool isCommand = false;
+  bool condition[RULES_IF_MAX_NESTING_LEVEL];
+  bool ifBranche[RULES_IF_MAX_NESTING_LEVEL];
+  byte ifBlock     = 0;
+  byte fakeIfBlock = 0;
+
+  std::vector<byte> buf;
```

Comment:
std::vector<byte> buf is better than byte array?

Reply:
The SPIFFS_CHECK can return the function, so there is no delete of the buf.
std::vector (like any other STL container) does clean up the allocated stuff when destructed.